### PR TITLE
D8CORE-3263 Gallery paragraph styling

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Styles for image gallery paragraph type
- Dynamic scss compiling
   - Now simply adding a new file will generated a compiled css file. no need to edit the webpack file. it will treat any file that begins with an underscore as a partial and not use that as an entry point.
- Dynamic library generation
   - Each css file will create a respective library. No more need to edit the libraries.yml file each time.

# Need Review By (Date)
- 1/14

# Urgency
- medium

# Steps to Test
1. checkout this branch
2. clear caches
3. verify nothing looks different.

Other steps will be separate in https://github.com/SU-SWS/stanford_profile/pull/347

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by jenbreese** [comment]
Is there another step besides the one in the other pull request? It doesn't really work on my build.  
image

**Comment by pookmish** [comment]
@jenbreese what doesn't work? your screenshot looks accurate.

**Comment by jenbreese** [comment]
But I though there was an option to have a scrolling moving gallery and then also a grid? On my local paragraph, I don't see an option to do that.

**Comment by pookmish** [comment]
The grid is the first phase. later we might implement the option to choose grid vs slideshow, but slideshows are not easy and so that'll be a later enhancement.

**Comment by jenbreese** [comment]
Ok. Great. My images are all the same size. That is expected? The Figma kind of looks like the images would be different sizes. 

I also noticed that the images don't wrap in the editor which looks a little funny.
Screen Shot 2021-01-12 at 1 51 37 PM

**Comment by pookmish** [comment]
Ishi and I decided we can review the image size at a later time. it might depend on real content.

I'm currently working on the react UI and should have that up soon.

**Comment by jenbreese** [comment]
I think the CSS looks good.  Thank you for clarifying.

File: modules/stanford_profile_styles/stanford_profile_styles.module [hunk 1]
```diff
@@ -34,18 +33,57 @@ function stanford_profile_styles_page_attachments(array &$attachments) {
...
+    // This is the directory the file lives in.
+    $bucket = reset($path_parts);
+    $lib = basename($file->getFilename(), '.css');
+
+    // Build the library definition.
+    $libraries[trim("$bucket.$lib", '. ')] = [
+      'css' => [
+        'component' => [
+          "$local_path/{$file->getFileName()}" => [],
+        ],
```
Comment (sherakama) [inline modules/stanford_profile_styles/stanford_profile_styles.module:79]: This is mighty slick but mighty assumptive. Component might not always be the best option for each css file. Perhaps you can come up with a naming convention that will allow for stylesheets to be placed at different levels? eg: `thing.base.css` or `thing.theme.css` and default to component if not specified.

File: modules/stanford_profile_styles/webpack.config.js [hunk 2]
```diff
@@ -38,25 +39,30 @@ const distJS = path.resolve(__dirname, distDir, "js");
...
+  if (fileName.indexOf('_') === 0) {
+    return acc;
+  }
+  if (fileName === 'index.scss') {
+    fileName = filePathParts.pop();
+  }
+  const entry = filePathParts.length >= 1 ? filePathParts.join('/') + '/' + fileName : fileName;
+  acc[entry.replace('.scss', '')] = filePath
+  return acc
+}, {});
```
Comment (sherakama) [inline modules/stanford_profile_styles/webpack.config.js:55]: Fancy 👍

File: modules/stanford_profile_styles/stanford_profile_styles.module (hunk 1 above)
Comment (pookmish) [reply modules/stanford_profile_styles/stanford_profile_styles.module:79]: hmm. that's a pretty neat idea. i struggled with trying to figure that out. i like that idea.

Comment (pookmish) [reply modules/stanford_profile_styles/stanford_profile_styles.module]: I created level directories to create these. should be pretty straight forward now.

**Review by sherakama — APPROVED** [review]
Love this. So tidy. Nice work.

**Pull request merged**